WordPress/Classes/Services/MediaSettings.swift

```diff
@@ -110,13 +151,19 @@ class MediaSettings: NSObject {
     /// - Note: if the image doesn't need to be resized, it returns `Int.max`
     ///
     @objc var imageSizeForUpload: Int {
-        if maxImageSizeSetting >= maxImageDimension {
+        // When image optimization is enabled, setting the max image size setting to
+        // the maximum value will be considered as to using the original size.
+        if !imageOptimizationEnabled || maxImageSizeSetting >= maxImageDimension {
             return Int.max
         } else {
             return maxImageSizeSetting
         }
     }
 
+    var imageQualityForUpload: ImageQuality {
+        return imageOptimizationEnabled ? imageQualitySetting : .high
```

Comment:
When image optimization is disabled, the quality used is `High` not `Maximum`. This behavior matches the previous implementation where `preferredImageCompressionQuality` was used:

https://github.com/wordpress-mobile/WordPress-iOS/blob/e7a35f173d73ef4c0f7e7089e6d2676433b9a4d9/WordPress/Classes/Services/MediaImportService.swift#L22

https://github.com/wordpress-mobile/WordPress-iOS/pull/21981/files#diff-c162ac46983f01e3b35e008aa8b553cc0cb2b010a84926da308ee9b875e35638L381